float64[] times     # N timestamps (in seconds)
float32[] gains     # N-by-UD-by-XD gain matrices
